commit 16e7070b64e4fe20d87c12ee806eb9201daba29d
Author: Catalin Patulea <cronos586@gmail.com>
Date:   Sat Jan 27 11:22:24 2024 +0000

    Revert "Delete setup.py."
    
    This reverts commit c9bb83f85c340aff02b87746a665375927cede2c.

diff --git a/setup.py b/setup.py
new file mode 100755
index 0000000..cbbbecb
--- /dev/null
+++ b/setup.py
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) Dnspython Contributors, see LICENSE for text of ISC license
+#
+# Copyright (C) 2003-2007, 2009-2011 Nominum, Inc.
+#
+# Permission to use, copy, modify, and distribute this software and its
+# documentation for any purpose with or without fee is hereby granted,
+# provided that the above copyright notice and this permission notice
+# appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND NOMINUM DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL NOMINUM BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT
+# OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+import sys
+
+from setuptools import setup
+
+try:
+    sys.argv.remove("--cython-compile")
+except ValueError:
+    compile_cython = False
+else:
+    compile_cython = True
+    from Cython.Build import cythonize
+
+    ext_modules = cythonize(
+        [
+            "dns/*.py",
+            "dns/dnssecalgs/*.py",
+            "dns/quic/*.py",
+            "dns/rdtypes/*.py",
+            "dns/rdtypes/*/*.py",
+        ],
+        language_level="3",
+    )
+
+kwargs = {
+    "ext_modules": ext_modules if compile_cython else None,
+    "zip_safe": False if compile_cython else None,
+}
+
+setup(**kwargs)
